dask/dataframe/core.py

```diff
@@ -283,48 +263,56 @@ def cache(self, cache=Cache):
         name = 'from-cache-' + self._name
         dsk2 = dict(((name, i), (getitem, cache, (tuple, list(key))))
                     for i, key in enumerate(self._keys()))
-        return self._constructor(dsk2, name, self._pd, self.divisions)
+        return self._constructor(dsk2, name, self._meta, self.divisions)
 
     @derived_from(pd.DataFrame)
     def drop_duplicates(self, **kwargs):
         assert all(k in ('keep', 'subset', 'take_last') for k in kwargs)
         chunk = lambda s: s.drop_duplicates(**kwargs)
-        return aca(self, chunk=chunk, aggregate=chunk, columns=self._pd,
+        return aca(self, chunk=chunk, aggregate=chunk, meta=self._meta,
                    token='drop-duplicates')
 
     def __len__(self):
         return reduction(self, len, np.sum, token='len').compute()
 
-    def map_partitions(self, func, columns=no_default, *args, **kwargs):
-        """ Apply Python function on each DataFrame block
-
-        When using ``map_partitions`` you should provide either the column
-        names (if the result is a DataFrame) or the name of the Series (if the
-        result is a Series).  The output type will be determined by the type of
-        ``columns``.
+    def map_partitions(self, func, *args, **kwargs):
+        """ Apply Python function on each DataFrame block.
 
         Parameters
         ----------
 
         func : function
             Function applied to each blocks
-        columns : tuple or scalar
-            Column names or name of the output. Defaults to names of data itself.
-            When tuple is passed, DataFrame is returned. When scalar is passed,
-            Series is returned.
+        args, kwargs :
+            Arguments and keywords to pass to the function. The partition will
+            be the first argument, and these will be passed *after*.
+        meta : pd.DataFrame, pd.Series, dict, tuple, optional
+            Metadata describing the output DataFrame or Series. Should have
```

Comment:
Suggest something like the following:

>  An empty pandas DataFrame or Series that matches the dtypes and column names of the output.  Alternatively ...

Leading with the term metadata might be confusing to newcomers.